cmd/kubectl-gadget/main.go

```diff
@@ -72,34 +77,45 @@ func main() {
 		}
 	}
 
-	runtime := grpcruntime.New(grpcruntime.WithConnectUsingK8SProxy)
-	runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
-	common.AddFlags(rootCmd, runtimeGlobalParams, nil, runtime)
-	runtime.Init(runtimeGlobalParams)
-	if !skipInfo {
-		// evaluate flags early for runtimeGlobalFlags; this will make
-		// sure that --connection-method has already been parsed when calling
-		// InitDeployInfo()
-
-		err := commonutils.ParseEarlyFlags(rootCmd, os.Args[1:])
-		if err != nil {
-			// Analogous to cobra error message
-			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
-			os.Exit(1)
+	grpcRuntime = grpcruntime.New(grpcruntime.WithConnectUsingK8SProxy)
+	runtimeGlobalParams = grpcRuntime.GlobalParamDescs().ToParams()
+	common.AddFlags(rootCmd, runtimeGlobalParams, nil, grpcRuntime)
+	grpcRuntime.Init(runtimeGlobalParams)
+
+	// evaluate flags early for runtimeGlobalParams; this will make
+	// sure that all flags relevant for the grpc connection are ready
+	// to be used
+
+	err := commonutils.ParseEarlyFlags(rootCmd, os.Args[1:])
+	if err != nil {
+		// Analogous to cobra error message
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	if !skipInfo && !runtimeGlobalParams.Get(grpcruntime.ParamGadgetNamespace).IsSet() {
```

Comment:
Any reason why not to enable the logic that detects the namespace for the deploy and version commands? IMO it makes sense to have it.